Runtime/Behaviours/CinemachineConfiner2D.cs

```diff
@@ -0,0 +1,479 @@
+    /// <item><description> when the input polygon is non-uniformly scaled, or.</description></item>
+    /// <item><description> when the input polygon is rotated.</description></item>
+    /// </list>
+    /// For efficiency reasons, Cinemachine will not automatically regenerate the inner polygon 
+    /// in these cases, and it is the responsibility of the client to call the InvalidateCache() 
+    /// method to trigger the recalculation. An inspector button is also provided for this purpose.
+    /// </para>
+    ///
+    /// <para>
+    /// If the input polygon scales uniformly or translates, the cache remains valid. If the 
+    /// polygon rotates, then the cache degrades in quality (more or less depending on the aspect 
+    /// ratio - it's better if the ratio is close to 1:1) but can still be used. 
+    /// Regenerating it will eliminate the imperfections.
+    /// </para>
+    ///
+    /// <para>
+    /// The cached secondary polygon is not a single polygon, but rather a family of polygons from 
+    /// which a member is chosen depending on the current size of the camera view. The number of 
+    /// polygons in this family will depend on the complexity of the input polygon, and the maximum 
+    /// expected camera view size. The MaxOrthoSize property is provided to give a hint to the 
+    /// algorithm to stop generating polygons for camera view sizes larger than the one specified. 
+    /// This can represent a substantial cost saving when regenerating the cache, so it is a good 
+    /// idea to set it carefully. Leaving it at 0 will cause the maximum number of polygons to be generated.
+    /// </para>
+    /// </summary>
+    [SaveDuringPlay, ExecuteAlways]
+    public class CinemachineConfiner2D : CinemachineExtension
+    {
+        /// <summary>The 2D shape within which the camera is to be contained.</summary>
+        [Tooltip("The 2D shape within which the camera is to be contained.  " +
+                 "Can be a 2D polygon or 2D composite collider.")]
+        public Collider2D m_BoundingShape2D;
+
+        /// <summary>Damping applied automatically around corners to avoid jumps.</summary>
+        [Tooltip("Damping applied around corners to avoid jumps.  Higher numbers are more gradual.")]
+        [Range(0, 5)]
+        public float m_Damping;
+
+        /// <summary>
+        /// The confiner will correctly confine up to the maximum window size (for orthographic camera, this is
+        /// equivalent to the orthographic size) or the first self intersection, whichever comes first.
+        /// If set to 0, then this parameter is ignored. Use it to optimize computation and memory costs.
+        /// </summary>
+        [Tooltip("The confiner will correctly confine up to the maximum window size (for orthographic camera, this is " +
+                 "equivalent to the orthographic size) or the first self intersection, whichever comes first. " +
```

Comment:
You can't talk about "self first intersection".  Nobody other then us will know what that means.  Just FYI.
I have adjusted this text and pushed it.  I also made the corresponding change the Bryan's manual doc.